components/formats-gpl/src/loci/formats/in/HarmonyReader.java

```diff
@@ -0,0 +1,783 @@
+  }
+
+  /* @see loci.formats.IFormatReader#isThisType(RandomAccessInputStream) */
+  @Override
+  public boolean isThisType(RandomAccessInputStream stream) throws IOException {
+    TiffParser p = new TiffParser(stream);
+    IFD ifd = p.getFirstIFD();
+    if (ifd != null) {
+      Object s = ifd.getIFDValue(XML_TAG);
+      if (s == null) return false;
+      String xml = s instanceof String[] ? ((String[]) s)[0] : s.toString();
+      return xml.indexOf(MAGIC) < 1024;
+    }
+    stream.seek(0);
+    String xml = stream.readString(1024);
+    return xml.indexOf(MAGIC) > 0;
+  }
+
+  /* @see loci.formats.IFormatReader#getUsedFiles(boolean) */
+  @Override
+  public String[] getUsedFiles(boolean noPixels) {
+    FormatTools.assertId(currentId, true, 1);
+
+    ArrayList<String> files = new ArrayList<String>();
+    files.addAll(metadataFiles);
+    if (!noPixels) {
+      for (Plane[] well : planes) {
+        for (Plane p : well) {
+          files.add(p.filename);
+        }
+      }
+    }
+    return files.toArray(new String[files.size()]);
+  }
+
+  /* @see loci.formats.IFormatReader#getSeriesUsedFiles(boolean) */
+  @Override
+  public String[] getSeriesUsedFiles(boolean noPixels) {
+    FormatTools.assertId(currentId, true, 1);
+
+    ArrayList<String> files = new ArrayList<String>();
+    files.addAll(metadataFiles);
+    if (!noPixels) {
+      for (Plane p : planes[getSeries()]) {
+        files.add(p.filename);
+      }
+    }
+
+    return files.toArray(new String[files.size()]);
+  }
+
+  /* @see loci.formats.IFormatReader#close(boolean) */
+  @Override
+  public void close(boolean fileOnly) throws IOException {
+    super.close(fileOnly);
+    if (!fileOnly) {
+      if (reader != null) {
+        reader.close();
+      }
+      reader = null;
+      planes = null;
+      metadataFiles.clear();
+    }
+  }
+
+  /**
+   * @see loci.formats.IFormatReader#openBytes(int, byte[], int, int, int, int)
+   */
+  @Override
+  public byte[] openBytes(int no, byte[] buf, int x, int y, int w, int h)
+    throws FormatException, IOException
+  {
+    FormatTools.checkPlaneParameters(this, no, buf.length, x, y, w, h);
+
+    if (getSeries() < planes.length && no < planes[getSeries()].length) {
+      Plane p = planes[getSeries()][no];
+
+      if (new Location(p.filename).exists()) {
+        if (reader == null) {
+          reader = new MinimalTiffReader();
+        }
+        reader.setId(p.filename);
+        reader.openBytes(0, buf, x, y, w, h);
+        reader.close();
+      }
+    }
+
+    return buf;
+  }
+
+  // -- Internal FormatReader API methods --
+
+  /* @see loci.formats.FormatReader#initFile(String) */
+  @Override
+  protected void initFile(String id) throws FormatException, IOException {
+    // make sure that we have the XML file and not a TIFF file
+
+    if (!checkSuffix(id, "xml")) {
+      Location parent = new Location(id).getAbsoluteFile().getParentFile();
+      Location xml = new Location(parent, XML_FILES[0]);
```

Comment:
Line 237 will only evaluate to ```true``` if one of the TIFF files is selected, which implies that ```Index.idx.xml``` is present and ```Index.ref.xml``` is not.  We can certainly change this later once we have a final decision on how to handle the ```Index.ref.xml``` files, but for now it seemed a safe assumption.